torch/serialization.py

```diff
@@ -1112,6 +1116,22 @@ def load_tensor(dtype, numel, key, location):
         name = f'data/{key}'
 
         storage = zip_file.get_storage_from_record(name, numel, torch.UntypedStorage)._typed_storage()._untyped_storage
+        # check if byteswapping is needed and swap here if it is needed
+        byteencodingname = 'byteencoding'
+        if zip_file.has_record(byteencodingname):
+            encodingdata = zip_file.get_record(byteencodingname)
+            need_byteswap = True
+            elem_size = torch._utils._element_size(dtype)
+            if elem_size == 1:
+                need_byteswap = False
+            if (encodingdata == b'le' and sys.byteorder == 'little') or (encodingdata == b'be' and sys.byteorder == 'big'):
+                need_byteswap = False
+            if need_byteswap:
+                # do byteswapping here
+                for i in range(int(len(storage) / elem_size)):
```

Comment:
I think there's something missing in the logic here. The swap should only happen for up to an 8 byte range, for dtypes larger than 8 bytes. For instance, if we have a 16 byte value, like a `torch.complex128`, the first set of 8 bytes will need to be swapped separately from the second set of 8 bytes. And I'm not actually sure how endianness affects complex dtypes in general

It would be nice to be able to reuse the byte swapping logic from `torch.UntypedStorage.from_buffer()`, but at the moment, that requires converting the data to a `bytes` object first...

Reply:
Yeah, it's definitely a good idea to check with complex types. Could you tell me how to modify code from #65300 to use complex types?

Reply:
This should work correctly now according to manual testing.